tests/Unit/model/CategoryTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Tests\Unit\Model;
+
+use App\Models\Category;
+use App\Models\Product;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\HasManyThrough;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Tests\TestCase;
+
+class CategoryTest extends TestCase
+{
+    /**
+     * A basic unit test example.
+     *
+     * @return void
+     */
+
+    public function testCategoryHasManyChildCategories()
+    {
+        $category = new Category();
+
+        $this->assertInstanceOf(HasMany::class, $category->childCategories());
+    }
+
+    public function testCategoryHasParentCategory()
+    {
+        $category = new Category();
+
+        $this->assertInstanceOf(BelongsTo::class, $category->parentCategory());
+    }
+
+    public function testCategoryHasManyChildProducts()
+    {
+        $category = new Category();
+
+        $this->assertInstanceOf(HasMany::class, $category->childproducts());
+    }
+
+    public function testCategoryHasManyGrandChildProducts()
+    {
+        $category = new Category();
+        
+        $this->assertInstanceOf(HasManyThrough::class, $category->grandchildproducts());
+    }
+
+    public function testFieldAreFillable()
+    {
+        $inputs = [
+            'name',
+            'slug',
+            'parent_id',
+        ];
+        $category =new Category();
```

Comment:
```suggestion
        $category = new Category();
```
function nào cũng new lại ntn sao ko dùng lun setUp tearDown nhỉ?